pyomo/contrib/fme/fourier_motzkin_elimination.py

```diff
@@ -313,14 +347,39 @@ def _fourier_motzkin_elimination(self, constraints, vars_to_eliminate):
                 # NOTE: neither of the scalar multiplications below flip the
                 # constraint. So we are sure to have only geq constraints
                 # forever, which is exactly what we want.
-                if leaving_var_coef < 0:
-                    leq_list.append(
-                        self._nonneg_scalar_multiply_linear_constraint(
-                            cons, -1.0/leaving_var_coef))
+                if not self.do_integer_arithmetic:
+                    if leaving_var_coef < 0:
+                        leq_list.append(
+                            self._nonneg_scalar_multiply_linear_constraint(
+                                cons, -1.0/leaving_var_coef))
+                    else:
+                        geq_list.append(
+                            self._nonneg_scalar_multiply_linear_constraint(
+                                cons, 1.0/leaving_var_coef))
                 else:
-                    geq_list.append(
-                        self._nonneg_scalar_multiply_linear_constraint(
-                            cons, 1.0/leaving_var_coef))
+                    if not _is_integer(leaving_var_coef):
```

Comment:
I think you actually need to convert x to an integer to get exact integer arithmetic.  Just checking that the value is integer without ensuring that the `type` is integer will still result in floating point operations.

Reply:
This one is I think okay because it is cast to an int in line 410. But I think you are still quite right, because I'm not making sure to cast the actual coefficients to integers before I start taking linear combinations of them...